### ISSUE TITLE
Aim 2.1.b. Provide reports regarding validation of KGX-format ontologies.

### ISSUE THREAD

**Issue by caufieldjh** [ncbo/BioPortal-to-KGX#27]
The goal stated in the title is more or less complete, so the remaining tasks here are:

- [x] Enumerate UMLS semantic types used in BioPortal
- [x] Determine extent of mapping between UMLS semantic types  and Biolink types
- [ ] Use results of KGX validations to determine where BioPortal type mappings may be improved, and in what way

**Comment by caufieldjh** [comment]
Conveniently enough, the Semantic Types are on BP: https://bioportal.bioontology.org/ontologies/STY

**Comment by caufieldjh** [comment]
**Ontologies with edges involving UMLS semantic types:**
ICD10PCS
ECP
COSTART
WHO-ART
ICPC2P
NDFRT
HCPCS
ICD9CM
ICD10
ROO
RXNORM
NCBITAXON
MDDB
MEDDRA
NDDF
HL7
SNMI
GSSO
MEDLINEPLUS
PDQ
RCD
CRISP
STY
OMIM
MSTDE
ATC
RO
MSTDE-FRE
AI-RHEUM
MESH
VANDF
ICD10CM

**Ontologies with nodes involving UMLS semantic types:**
ICD10PCS
ECP
COSTART
WHO-ART
ICPC2P
BIOLINK
NDFRT
HCPCS
ID-AMR
ICD9CM
CVAO
ICD10
ROO
RXNORM
NCBITAXON
MDDB
MEDDRA
NDDF
HL7
TXPO
SNMI
GSSO
HOIP
MEDLINEPLUS
PDQ
RCD
CRISP
STY
OMIM
MSTDE
STO-DRAFT
ATC
RO
MSTDE-FRE
AI-RHEUM
MESH
VANDF
ICD10CM

**Comment by caufieldjh** [comment]
**The following codes are used, followed by their usage (combined nodes+edges):**
1020785 http://purl.bioontology.org/ontology/STY/T204
 451099 http://purl.bioontology.org/ontology/STY/T007
 234635 http://purl.bioontology.org/ontology/STY/T002
 218628 http://purl.bioontology.org/ontology/STY/T061
 214615 http://purl.bioontology.org/ontology/STY/T004
 195986 http://purl.bioontology.org/ontology/STY/T109
 137735 http://purl.bioontology.org/ontology/STY/T200
 137667 http://purl.bioontology.org/ontology/STY/T121
 125187 http://purl.bioontology.org/ontology/STY/T116
 114231 http://purl.bioontology.org/ontology/STY/T047
  91456 http://purl.bioontology.org/ontology/STY/T037
  81193 http://purl.bioontology.org/ontology/STY/T033
  68658 http://purl.bioontology.org/ontology/STY/T123
  52077 http://purl.bioontology.org/ontology/STY/T013
  48468 http://purl.bioontology.org/ontology/STY/T028
  45631 http://purl.bioontology.org/ontology/STY/T005
  29269 http://purl.bioontology.org/ontology/STY/T126
  29003 http://purl.bioontology.org/ontology/STY/T060
  26791 http://purl.bioontology.org/ontology/STY/T074
  25200 http://purl.bioontology.org/ontology/STY/T191
  21837 http://purl.bioontology.org/ontology/STY/T046
  16682 http://purl.bioontology.org/ontology/STY/T129
  15662 http://purl.bioontology.org/ontology/STY/T023
  15172 http://purl.bioontology.org/ontology/STY/T012
  12528 http://purl.bioontology.org/ontology/STY/T014
  12490 http://purl.bioontology.org/ontology/STY/T015
  12060 http://purl.bioontology.org/ontology/STY/T019
  11692 http://purl.bioontology.org/ontology/STY/T011
   9902 http://purl.bioontology.org/ontology/STY/T114
   9815 http://purl.bioontology.org/ontology/STY/T130
   8996 http://purl.bioontology.org/ontology/STY/T184
   8466 http://purl.bioontology.org/ontology/STY/T059
   8031 http://purl.bioontology.org/ontology/STY/T194
   7765 http://purl.bioontology.org/ontology/STY/T048
   6444 http://purl.bioontology.org/ontology/STY/T122
   6105 http://purl.bioontology.org/ontology/STY/T170
   5987 http://purl.bioontology.org/ontology/STY/T197
   5770 http://purl.bioontology.org/ontology/STY/T195
   5753 http://purl.bioontology.org/ontology/STY/T168
   5510 http://purl.bioontology.org/ontology/STY/T058
   5507 http://purl.bioontology.org/ontology/STY/T131
   5485 http://purl.bioontology.org/ontology/STY/T192
   4942 http://purl.bioontology.org/ontology/STY/T097
   3941 http://purl.bioontology.org/ontology/STY/T125
   3938 http://purl.bioontology.org/ontology/STY/T190
   3554 http://purl.bioontology.org/ontology/STY/T020
   3263 http://purl.bioontology.org/ontology/STY/T034
   3160 http://purl.bioontology.org/ontology/STY/T073
   2920 http://purl.bioontology.org/ontology/STY/T081
   2877 http://purl.bioontology.org/ontology/STY/T203
   2803 http://purl.bioontology.org/ontology/STY/T029
   2802 http://purl.bioontology.org/ontology/STY/T196
   2477 http://purl.bioontology.org/ontology/STY/T169
   2416 http://purl.bioontology.org/ontology/STY/T042
   2372 http://purl.bioontology.org/ontology/STY/T098
   2274 http://purl.bioontology.org/ontology/STY/T127
   2154 http://purl.bioontology.org/ontology/STY/T082
   2078 http://purl.bioontology.org/ontology/STY/T093
   1984 http://purl.bioontology.org/ontology/STY/T030
   1810 http://purl.bioontology.org/ontology/STY/T044
   1694 http://purl.bioontology.org/ontology/STY/T080
   1583 http://purl.bioontology.org/ontology/STY/T078
   1487 http://purl.bioontology.org/ontology/STY/T025
   1293 http://purl.bioontology.org/ontology/STY/T083
   1023 http://purl.bioontology.org/ontology/STY/T031
   1017 http://purl.bioontology.org/ontology/STY/T067
    999 http://purl.bioontology.org/ontology/STY/T079
    999 http://purl.bioontology.org/ontology/STY/T041
    999 http://purl.bioontology.org/ontology/STY/T026
    992 http://purl.bioontology.org/ontology/STY/T171
    873 http://purl.bioontology.org/ontology/STY/T024
    850 http://purl.bioontology.org/ontology/STY/T167
    850 http://purl.bioontology.org/ontology/STY/T040
    832 http://purl.bioontology.org/ontology/STY/T185
    819 http://purl.bioontology.org/ontology/STY/T039
    763 http://purl.bioontology.org/ontology/STY/T049
    760 http://purl.bioontology.org/ontology/STY/T070
    685 http://purl.bioontology.org/ontology/STY/T091
    685 http://purl.bioontology.org/ontology/STY/T055
    623 http://purl.bioontology.org/ontology/STY/T054
    622 http://purl.bioontology.org/ontology/STY/T043
    595 http://purl.bioontology.org/ontology/STY/T057
    571 http://purl.bioontology.org/ontology/STY/T056
    543 http://purl.bioontology.org/ontology/STY/T201
    539 http://purl.bioontology.org/ontology/STY/T032
    526 http://purl.bioontology.org/ontology/STY/T120
    512 http://purl.bioontology.org/ontology/STY/T062
    506 http://purl.bioontology.org/ontology/STY/T104
    466 http://purl.bioontology.org/ontology/STY/T018
    462 http://purl.bioontology.org/ontology/STY/T090
    426 http://purl.bioontology.org/ontology/STY/T099
    425 http://purl.bioontology.org/ontology/STY/T077
    405 http://purl.bioontology.org/ontology/STY/T068
    391 http://purl.bioontology.org/ontology/STY/T045
    384 http://purl.bioontology.org/ontology/STY/T065
    369 http://purl.bioontology.org/ontology/STY/T022
    339 http://purl.bioontology.org/ontology/STY/T052
    312 http://purl.bioontology.org/ontology/STY/T092
    304 http://purl.bioontology.org/ontology/STY/T064
    283 http://purl.bioontology.org/ontology/STY/T001
    282 http://purl.bioontology.org/ontology/STY/T038
    282 http://purl.bioontology.org/ontology/STY/T010
    277 http://purl.bioontology.org/ontology/STY/T101
    271 http://purl.bioontology.org/ontology/STY/T063
    258 http://purl.bioontology.org/ontology/STY/T096
    254 http://purl.bioontology.org/ontology/STY/T008
    248 http://purl.bioontology.org/ontology/STY/T072
    234 http://purl.bioontology.org/ontology/STY/T021
    233 http://purl.bioontology.org/ontology/STY/T017
    205 http://purl.bioontology.org/ontology/STY/T089
    188 http://purl.bioontology.org/ontology/STY/T053
    182 http://purl.bioontology.org/ontology/STY/T085
    181 http://purl.bioontology.org/ontology/STY/T103
    178 http://purl.bioontology.org/ontology/STY/T102
    177 http://purl.bioontology.org/ontology/STY/T100
    175 http://purl.bioontology.org/ontology/STY/T086
    167 http://purl.bioontology.org/ontology/STY/T069
    166 http://purl.bioontology.org/ontology/STY/T066
    161 http://purl.bioontology.org/ontology/STY/T051
    157 http://purl.bioontology.org/ontology/STY/T050
    147 http://purl.bioontology.org/ontology/STY/T087
    129 http://purl.bioontology.org/ontology/STY/T071
    124 http://purl.bioontology.org/ontology/STY/T075
    115 http://purl.bioontology.org/ontology/STY/T095
    105 http://purl.bioontology.org/ontology/STY/T094
     98 http://purl.bioontology.org/ontology/STY/T016
     91 http://purl.bioontology.org/ontology/STY/T088
      5 http://purl.bioontology.org/ontology/STY/T119
      3 http://purl.bioontology.org/ontology/STY/T124
      3 http://purl.bioontology.org/ontology/STY/T118
      3 http://purl.bioontology.org/ontology/STY/T115
      3 http://purl.bioontology.org/ontology/STY/T111
      3 http://purl.bioontology.org/ontology/STY/T110

**Comment by caufieldjh** [comment]
Mapping between STY and Biolink as per LOOM is limited to 9 pairs.
See https://github.com/ncbo/BioPortal-to-KGX/blob/27-aim-21b-provide-reports-regarding-validation-of-kgx-format-ontologies/mappings/sty-biolink-exact-0.1.sssom.tsv for these lexical maps and a further set of curated/logical maps.

**Comment by cmungall** [comment]
https://github.com/biolink/biolink-model/blob/d050b64ff28372488f05e03df2829c461998809a/biolink-model.yaml#L6554

**Comment by caufieldjh** [comment]
The following ontologies use UMLS semantic types in nodes and/or edges but aren't directly imported by [Bioportal's UMLS ingest](https://github.com/ncbo/umls2rdf):
ECP
ROO
GSSO
STY
RO
BIOLINK
ID-AMR
CVAO
TXPO
HOIP
STO-DRAFT

**Comment by caufieldjh** [comment]
> https://github.com/biolink/biolink-model/blob/d050b64ff28372488f05e03df2829c461998809a/biolink-model.yaml#L6554

I combined the Biolink STY maps with additional maps I curated where they didn't yet exist. Immediate take-homes, most if not all of them known properties of Biolink:
* STY Levels of taxonomy don't have more specific corresponding classes than `biolink:OrganismTaxon` - could make it challenging to specify taxon-specific relationships
* Anatomy doesn't get into much detail in either Biolink or STY - there are better options for getting into rigorous detail about this - but Biolink doesn't specify much difference between a tissue, an organ, or other anatomy. Potentially not an issue.
* Biolink doesn't specify extensive detail re: varying chemical or molecule types, e.g. lipids or carbohydrates
